### DISCUSSION TITLE
What is the best way to derive FAIR metrics from a LinkML schema?

### DISCUSSION THREAD

**Discussion by cmungall** [linkml/linkml#1811] [Q&A]
There is a proliferation of FAIR checklists. These usually involve someone going through and manually answering questions on a form, enumerating questions about things like which standards are followed. This is often time-consuming and usually incomplete for large projects because it's hard for one person to hold all this in their head - and likely to become out of date.

E.g. [FIP](https://www.go-fair.org/how-to-go-fair/fair-implementation-profile/) has the following Qs:

![image](https://github.com/linkml/linkml/assets/50745/7a8611ca-a166-41ef-91d7-148f19ed77e6)

Some questions like I2 can be answered programmatically, looking at [uris and mappings](https://linkml.io/linkml/schemas/uris-and-mappings.html) as well as [value sets/enums](https://linkml.io/linkml/schemas/enums.html). Others may be out of band for the schema, although in some cases we may want to encourage more schema-level metadata that describes the information ecosystem the schema is intended for.

F1 could be answered by introspecting whether `identifier` is declared, what the range is, and what constraints are on that range.

It looks like FIP has an RDF serialization, in principle it should be possible to have a simple linkml tool that would partially populate a FIP - or perhaps just a TSV output is sufficient.

I think this would be beneficial as it would lower the barrier to continuous evaluation of FAIRness metrics, rather than this be a one-off box checking exercise that rapidly becomes incomplete and stale.

**Comment by jeffhhk** [comment]
As a stab at what might not be automatically derivable, the ShEx spec refers to a FAIRDataTeam schema for FAIR.  I'm not sure how complete it is intending to be in comparison to the resource you name.
> https://github.com/shexSpec/schemas/tree/master/FAIR
> https://github.com/FAIRDataTeam/metadata-schema
